raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -193,21 +193,18 @@ private void updateLeaderEndOffsetAndTimestamp(LeaderState state) {
         maybeUpdateFetchTimerWithRemoteFetchTimestamp(state, state.localId(), time.milliseconds());
     }
 
-    private void maybeUpdateFetchTimerWithRemoteFetchTimestamp(LeaderState state, int replicaId, long timestamp) {
-        final OptionalLong lastFetchTimestamp = state.updateFetchTimestamp(replicaId, timestamp);
-        if (lastFetchTimestamp.isPresent()) {
-            timer.resetDeadline(lastFetchTimestamp.getAsLong() + fetchTimeoutMs);
-        }
-    }
-
-    private void updateReplicaEndOffset(LeaderState state, int replicaId, long endOffset, long timestamp) {
+    private void updateReplicaEndOffset(LeaderState state, int replicaId, long endOffset) {
         if (state.updateEndOffset(replicaId, endOffset)) {
             logger.debug("Leader high watermark updated to {} after replica {} end offset updated to {}",
                     state.highWatermark(), replicaId, endOffset);
             applyCommittedRecordsToStateMachine();
         }
 
         // maybe extend the fetch timer with the majority of voter-fetching timestamps
+        maybeUpdateFetchTimerWithRemoteFetchTimestamp(state, replicaId, time.milliseconds());
+    }
+
+    private void maybeUpdateFetchTimerWithRemoteFetchTimestamp(LeaderState state, int replicaId, long timestamp) {
```

Comment:
nit: I guess you could go one step further and remove `timestamp` here

Reply:
Ack.